packages/teleport/src/DesktopSession/DesktopSession.tsx

```diff
@@ -229,16 +238,20 @@ function Session(props: PropsWithChildren<State>) {
     clipboardSuccess;
 
   const onShareDirectory = () => {
-    window
-      .showDirectoryPicker()
-      .then(sharedDirHandle => {
-        setIsSharingDirectory(true);
-        tdpClient.addSharedDirectory(sharedDirHandle);
-        tdpClient.sendSharedDirectoryAnnounce();
-      })
-      .catch(() => {
-        setIsSharingDirectory(false);
-      });
+    try {
+      window
+        .showDirectoryPicker()
+        .then(sharedDirHandle => {
+          setIsSharingDirectory(true);
+          tdpClient.addSharedDirectory(sharedDirHandle);
+          tdpClient.sendSharedDirectoryAnnounce();
+        })
+        .catch(() => {
+          setIsSharingDirectory(false);
+        });
+    } catch (e) {
```

Comment:
I'm not used to seeing a promise's `.catch` used in conjunction with a standard catch block. Are both of them going to be called in the event of an error?

Reply:
> Are both of them going to be called in the event of an error?

No, what this means is if the `Promise` returned by `window.showDirectoryPicker()` is rejected (which happens if the user closes the directory picker without selecting a directory), we `setIsSharingDirectory(false);`. On the other hand, if the call to `window.showDirectoryPicker()` throws an error (as happens on browsers where this feature is not implemented, i.e.:`Uncaught TypeError: window.showDirectoryPicker is not a function`), we `setDirectorySharingBrowserErr(true);`.

Granted I'm making the assumption that those are the only two situations common enough to distinguish between, and we don't need to implement any finer grained handling within each block. In theory it's possible that there's some other reason that the promise might be rejected, and some other error that could be thrown, in which case the behavior may or may not be what we want.